pydmd/rdmd.py

```diff
@@ -0,0 +1,126 @@
+"""
+Derived module from cdmd.py for Randomized DMD
+
+Reference:
+N. Benjamin Erichson, Lionel Mathelin, J. Nathan Kutz, Steven L. Brunton.
+Randomized dynamic mode decomposition. SIAM Journal on Applied Dynamical
+Systems, 18, 2019.
+"""
+
+import numpy as np
+from .cdmd import CDMD
+from .utils import compute_tlsq
+
+def randomized_qb(X, target_rank, oversampling, power_iters):
+    """
+    Randomized QB Decomposition.
+
+    :param numpy.ndarray X: (n, m) matrix to decompose.
+    :param int target_rank: Target rank << min(m, n) of the matrix X.
+    :param int oversampling: Number of additional samples to use when
+        computing the random test matrix.
+        Note that oversampling = {5,10} is often sufficient.
+    :param int power_iters: Number of power iterations to perform.
+        Note that power_iters = {1,2} leads to considerable improvements.
+
+    :return: a near-optimal orthonormal basis Q for the range of the matrix X,
+        and the projection B of the matrix X onto the low-dimensional space Q.
+    :rtype: (n, target_rank) numpy.ndarray, (target_rank, m) numpy.ndarray
+
+    Reference:
+    N. Benjamin Erichson, Lionel Mathelin, J. Nathan Kutz, Steven L. Brunton.
+    Randomized dynamic mode decomposition. SIAM Journal on Applied Dynamical
+    Systems, 18, 2019.
+    """
+    m = X.shape[1]
+
+    # Generate random test matrix (with slight oversampling)
+    Omega = np.random.randn(m, target_rank + oversampling)
+
+    # Compute sampling matrix
+    Y = X.dot(Omega)
+
+    # Perform power iterations
+    for j in range(power_iters):
```

Comment:
Apparently `j` is not used anywhere in the loop, so you could just replace it with `_` (this fixes a Codacy warning).

```suggestion
    for _ in range(power_iters):
```